### PR TITLE
SQL comments to index positional arguments

### PR DESCRIPTION
**Pull request by seancolsen**
This PR is a small non-functional change to serve as a sort of async discussion thread.

@mathemancer and @Anish9901, what would your appetite be for allowing some manually-maintained comments in our SQL codebase like this? I recognize that they're more work to maintain. But, being that code gets read more than written, I would find comments like this quite helpful. Personally, I don't think I've ever worked with such poorly readable code in my whole career. Like, I cannot believe I need to _manually count over a dozen expressions_ just to read something. This is not to blame anyone who wrote this particular code, but rather just a general frustration with the limited expressiveness of PL/pgSQL.

With the comments there, I can highlight a positional index and then see all the places it occurs like this:

image

But if you don't like the idea of these comment indexes, then I won't make a stink about it, and we can close this PR. Or, if you like this idea, then we can merge this and add them as needed.

I'm not proposing that we use them everywhere. And I'm not proposing that we do any additional work to go add them all over the place right now. I'd just like to be able to add them to code blocks like this which are particularly unwieldy.

I have a VS Code extension called vscode-input-sequence which makes adding and updating sequences like this pretty easy. Perhaps theres Emacs as something like this too.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Anish9901** [comment]
I do like the idea @seancolsen, but I prefer the following style slightly more that the one in this PR.
https://github.com/mathesar-foundation/mathesar/blob/207eb7e18b9f0b4ffbebe31120717bea8df33d85/db/sql/00_msar.sql#L4073-L4079

File: db/sql/00_msar.sql [hunk 1]
```diff
@@ -4852,21 +4852,27 @@ BEGIN
...
-    msar.build_summary_cte_expr_for_table(tab_id),
-    msar.build_summary_join_expr_for_table(tab_id, 'enriched_results_cte'),
-    COALESCE(msar.build_summary_json_expr_for_table(tab_id), 'NULL'),
-    COALESCE(
+    /* %1 */ msar.build_selectable_column_expr(tab_id),
+    /* %2 */ msar.get_relation_schema_name(tab_id),
+    /* %3 */ msar.get_relation_name(tab_id),
+    /* %4 */ limit_,
+    /* %5 */ offset_,
+    /* %6 */ msar.build_order_by_expr(tab_id, order_),
```
Comment (mathemancer) [inline db/sql/00_msar.sql:4860]:
```suggestion
    msar.build_selectable_column_expr(tab_id), -- %1
    msar.get_relation_schema_name(tab_id), -- %2
    msar.get_relation_name(tab_id), -- %3
    limit_, -- %4
    offset_, -- %5
    msar.build_order_by_expr(tab_id, order_), -- %6
```

**Review by mathemancer — COMMENTED** [review]
Love this idea.

I really wish the `format` function had some way to give named parameters!

I made another suggestion for format of the comments.

My reasoning is:
- We should almost never need more explanation (since the functions should already have documentation and reasonable names)
- We should _never_ add the formatting character to the comment as @Anish9901 suggested, since it's not guaranteed to be consistent in the formatting string
- I like suffix rather than prefix since it's easier to ignore if you don't need it, and it's (slightly) more compact.

Long term, I had a project rattling around somewhere in my head that we should write an alternate `format_kwargs` function that did take named parameters, but @seancolsen 's solution is a whole lot easier and more likely to happen soon.

**Comment by seancolsen** [comment]
We discussed this during our team meeting today and decided to go with my proposed prefix approach. As a result, I'm merging this now.

**Pull request merged**